packages/pagination/Pagination.tsx

```diff
@@ -0,0 +1,276 @@
+export const INITIAL_PAGE_LENGTH = DEFAULT_PAGE_LENGTHS[1];
+const LARGE_PAGE_LENGTH_THRESHOLD = 500;
+const DEFAULT_ITEMS_LABEL = "Items";
+
+export const getItemCountString = (
+  activePage: number,
+  itemsLabel: string,
+  pageLength: number,
+  totalItems?: number
+) => {
+  const startItemIndex = (activePage - 1) * pageLength + 1;
+  const endItemIndex = totalItems
+    ? Math.min(activePage * pageLength, totalItems)
+    : activePage * pageLength;
+  const itemRange = `${startItemIndex}–${endItemIndex}`;
+  const totalItemsLabel = totalItems ? `${totalItems} ${itemsLabel}` : "";
+
+  if (!Boolean(totalItemsLabel)) {
+    return itemRange;
+  }
+
+  return `${itemRange} of ${totalItemsLabel}`;
+};
+
+const getItemCountClassName = (showPageLengthMenu?: boolean) => {
+  if (showPageLengthMenu) {
+    return cx(border("horiz"), padding("horiz"), margin("horiz"));
+  }
+
+  return cx(border("right"), padding("right"), margin("right"));
+};
+
+const NavButton: React.FC<Partial<
+  React.ButtonHTMLAttributes<HTMLButtonElement>
+> &
+  Partial<ExpandedLinkProps> & { direction: "prev" | "next" }> = ({
+  direction,
+  disabled,
+  url,
+  ...other
+}) => {
+  delete other.children;
+
+  const ariaLabel = direction === "prev" ? "previous page" : "next page";
+  const children = (
+    <Icon
+      shape={
+        direction === "prev" ? SystemIcons.CaretLeft : SystemIcons.CaretRight
+      }
+      color={disabled ? themeBgDisabled : "inherit"}
+    />
+  );
+
+  return (
+    <Delegate
+      to={url ? ResetLink : ResetButton}
+      props={
+        url
+          ? {
+              ["aria-label"]: ariaLabel,
+              children,
+              url: disabled ? url : undefined,
+              ...(disabled ? { ariaDisabled: "true" } : {}),
+              ...other
+            }
+          : { ["aria-label"]: ariaLabel, children, disabled, ...other }
+      }
+    />
+  );
+};
+
+const Pagination: React.FC<PaginationProps> = ({
+  activePage: activePageProp,
+  initialActivePage = 1,
+  itemsLabel = DEFAULT_ITEMS_LABEL,
+  pageLength: pageLengthProp,
+  initialPageLength = INITIAL_PAGE_LENGTH,
+  nextUrl,
+  onChange,
+  prevUrl,
+  showPageLengthMenu,
+  totalItems,
+  totalPages: totalPagesProp
+}) => {
+  const [pageLengthMenuId] = useId(1, "pageLengthMenu");
+  const [activePageState, setActivePageState] = React.useState<number>(
+    initialActivePage
+  );
+  const [pageInputVal, setPageInputVal] = React.useState<number>(
+    initialActivePage
+  );
+  const [pageLengthState, setPageLengthState] = React.useState<number>(
+    initialPageLength
+  );
+
+  const activePage = activePageProp || activePageState;
+  const pageLength = pageLengthProp || pageLengthState;
+  const isTotalPagesCalculable = Boolean(totalItems && pageLength);
+  const calculatedTotalPages =
+    isTotalPagesCalculable && totalItems
+      ? Math.ceil(totalItems / pageLength)
+      : undefined;
+  const totalPages =
+    !totalPagesProp && isTotalPagesCalculable
```

Comment:
Why do we need a `totalPagesProp`? (in other words, could we remove it and always just calculate it based off totalItems & pageLength)?